test/terra/backends/test_pulse_simulator.py

```diff
@@ -0,0 +1,696 @@
+    # Test gates (using meas level 2)
+    # ---------------------------------------------------------------------
+    def test_x_gate(self):
+        """Test x gate. Set omega_d0=omega_0 (drive on resonance), phi=0, omega_a = pi/time
+        """
+
+        # set variables
+
+        # set omega_0, omega_d0 equal (use qubit frequency) -> drive on resonance
+        omega_0 = 2*np.pi*self.freq_qubit_0
+        omega_d0 = omega_0
+
+        # Require omega_a*time = pi to implement pi pulse (x gate)
+        # num of samples gives time
+        omega_a = np.pi/self.drive_samples
+
+        phi = 0
+
+        x_qobj = self.create_qobj(shots=256, omega_0=omega_0, omega_a=omega_a, omega_d0=omega_d0,
+                                  phi=phi, meas_level=2)
+        result = self.backend_sim.run(x_qobj).result()
+        counts = result.get_counts()
+
+        exp_result = {'1':256}
+
+        self.assertDictAlmostEqual(counts, exp_result)
+
+    def test_hadamard_gate(self):
+        """Test Hadamard. Is a rotation of pi/2 about the y-axis. Set omega_d0=omega_0
+        (drive on resonance), phi=-pi/2, omega_a = pi/2/time
+        """
+
+        # set variables
+
+        # set omega_0, omega_d0 equal (use qubit frequency) -> drive on resonance
+        omega_0 = 2*np.pi*self.freq_qubit_0
+        omega_d0 = omega_0
+
+        # Require omega_a*time = pi/2 to implement pi/2 rotation pulse
+        # num of samples gives time
+        omega_a = np.pi/2/self.drive_samples
+
+        phi = -np.pi/2
+        shots = 100000 # large number of shots so get good proportions
+
+        had_qobj = self.create_qobj(shots=shots, omega_0=omega_0, omega_a=omega_a,
+                                    omega_d0=omega_d0, phi=phi, meas_level=2)
+        result = self.backend_sim.run(had_qobj).result()
+        counts = result.get_counts()
+
+        # compare prop
+        prop = {}
+        for key in counts.keys():
+            prop[key] = counts[key]/shots
+
+        exp_prop = {'0':0.5, '1':0.5}
+
+        self.assertDictAlmostEqual(prop, exp_prop, delta=0.01)
+
+    def test_arbitrary_gate(self):
+        """Test a few examples w/ arbitary drive, phase and amplitude. """
+
+        # Gate 1
+
+        # set variables
+        omega_0 = 2*np.pi*self.freq_qubit_0
+        omega_d0 = omega_0+0.01
+
+        omega_a = 2*np.pi/3/self.drive_samples
+
+        phi = 5*np.pi/7
+        shots = 10000000 # large number of shots so get good proportions
+
+        # Run qobj and compare prop to expected result
+        qobj1 = self.create_qobj(shots=shots, omega_0=omega_0, omega_a=omega_a, omega_d0=omega_d0,
+                                 phi=phi, meas_level=2)
+        result1 = self.backend_sim.run(qobj1).result()
+        counts1 = result1.get_counts()
+
+        prop1 = {}
+        for key in counts1.keys():
+            prop1[key] = counts1[key]/shots
+
+        exp_prop1 = {'0':0.315253, '1':1-0.315253}
```

Comment:
Where do these values come from? If it's a closed-form analytical could you include that as a function and call it?